src/coverlet.core/Instrumentation/ModuleTrackerTemplate.cs

```diff
@@ -72,6 +73,8 @@ public static void RecordSingleHit(int hitLocationIndex)
 
         public static void UnloadModule(object sender, EventArgs e)
         {
+            var fileSystem = (IFileSystem)DependencyInjection.Current.GetService(typeof(IFileSystem));
```

Comment:
Revert update in this file please.
This is a key piece of instrumentation, this class is injected inside users dll so we have to try to pollute less possible and use "runtime" only class.
In this case is not a problem..I mean core lib and instrumented lib are in same directory so we don't have issue with load...but I don't know how this evolve in future so I prefer keep this class(tracker) "light".

Reply:
FYI I'm trying to get logs also for it https://github.com/tonerdo/coverlet/pull/553
And the idea si to enable instrumentation throught enviroment variables, this is the only static part needed today.

Reply:
Ok done, reverted the changes.